src/main/java/com/conveyal/analysis/components/broker/Broker.java

```diff
@@ -100,8 +100,11 @@ public interface Config {
     private final ListMultimap<WorkerCategory, Job> jobs =
             MultimapBuilder.hashKeys().arrayListValues().build();
 
-    /** The most tasks to deliver to a worker at a time. */
-    public final int MAX_TASKS_PER_WORKER = 16;
+    /**
+     * The most tasks to deliver to a worker at a time. Workers may request less tasks than this, and the broker should
+     * never send more than the minimum of the two values. 50 tasks gives response bodies of about 65kB.
+     */
+    public final int MAX_TASKS_PER_WORKER = 100;
```

Comment:
What is the justification for increasing to 100 from 16? Are there any other ramifications other than response size that we should be concerned about?

Reply:
Previously this determined a the amount of tasks that were regularly delivered to a worker. The worker couldn't apply nuanced backpressure to the stream, so we'd usually hand out the maximum number to any worker, as long as the job had that many tasks left unfinished. With the workers now stating how many tasks they want, this value functions more as a true upper limit than a typical value. The worker has its own internal task buffer based on how many processor cores it has. It only requests as many tasks as it needs to top up that buffer, and it does so often. So most of the time it is expected to request less than 100, and maybe even less than the 16 that were delivered in the past. But if it's really burning through tasks quickly and has 90 free slots, it can ask for 90 and get them. This then serves as a guard rail to keep the response size reasonable.

Reply:
Thanks for the explanation! 🙏